libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -177,7 +176,7 @@ internal class MapboxTripSession(
         }
         tripService.stopService()
         stopLocationUpdates()
-        navigatorJobController.job.cancelChildren()
```

Comment:
When thinking about adding a function to cancel the navigator dispatcher (e.g., navigator.cancelDispatcherJobs())

Initial preference is to not add it, because the navigator should never leak a forever running task. I think we should ensure these are all computation tasks. If the navigator needs to make a network request, they should not block this thread.